templates/base/head_script.tmpl

```diff
@@ -7,6 +7,7 @@ If you introduce mistakes in it, Gitea JavaScript code wouldn't run correctly.
 	window.addEventListener('error', function(e) {window._globalHandlerErrors=window._globalHandlerErrors||[]; window._globalHandlerErrors.push(e);});
 	window.config = {
 		appVer: '{{AppVer}}',
+		appVerMd5: '{{MD5 AppVer}}',
```

Comment:
I do not think here need another variable or MD5 hash (more CPU time).

`res.js?v=${appVer}` should do the work and is clear.

Reply:
agh! I missed this!

Agreed with @wxiaoguang here. 

This MD5 is being recalced every time this page is requested - this is a bad idea !

Reply:
`MD5 AppVer` is used in multiple places in HTML already for cache busting, so it's already calculated like 4-5 times every page load. I think if we are to optimize ist, it should be calculated once at app startup and then cached.

Reply:
Yup, it's better to remove all `MD5 AppVer` but it's a little out of scope so I didn't mention it (I have removed it and the `MD5` function in another discarded PR).

For this PR, I think there is no need to introduce more MD5 calculations. Others `MD5` can be removed in this PR together at this time or be done in the future.

Reply:
ps: I also think using `index.js?v=1.17.1` (`{{AppVer}}`) is clearer than `index.js?v=md5hash`(`{{MD5 AppVer}}`), when users report problems, it's clear about which version they are using.

Reply:
Version is not optimal in any case, so it should probably be changed to a content hash at some point.

Reply:
Also, I imagine escaping a string could be more processing than calculating an md5, so it's really a weak argument.

Reply:
> Version is not optimal in any case, so it should probably be changed to a content hash at some point. Plain version also needs url-escaping I think, so still some processing involved for the escaping.

I do not see why it's not optimal. While content hash also works, either LGTM. The Go template will do the escaping automatically. And JS can also handle the escaping correctly. Even if without escaping, the chars in "AppVersion" are not dangerous, the only ambiguity char plus `+` won't cause any problem even if it is left as it is.

> Also, I imagine escaping a string could be more processing than calculating an md5, so it's really a weak argument.

That's just imagine.

```go
func BenchmarkUrlEscape(b *testing.B) {
	for i := 0; i < b.N; i++ {
		url.QueryEscape("1.17.1")
	}
}
func BenchmarkMd5(b *testing.B) {
	for i := 0; i < b.N; i++ {
		md5.Sum([]byte("1.17.1"))
	}
}
```

```
cpu: Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
BenchmarkUrlEscape
BenchmarkUrlEscape-12    	56735932	        19.01 ns/op
BenchmarkMd5
BenchmarkMd5-12          	10553167	       108.7 ns/op
```